patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -292,21 +296,63 @@ public Recipe AddCondition(IEnumerable<Condition> conditions)
 		return this;
 	}
 
-	public Recipe DisableShimmer()
+	/// <summary>
+	/// Sets a condition delegate that will determine whether or not the recipe can be decrafted/changed in shimmer.. The condition can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="condition">The predicate delegate condition.</param>
+	/// <param name="description">A description of this condition. Use NetworkText.FromKey, or NetworkText.FromLiteral for this.</param>
+	public Recipe AddShimmerCondition(NetworkText description, Predicate<Recipe> condition) => AddShimmerCondition(new Condition(description, condition));
+
+	/// <summary>
+	/// Adds an array of conditions that will determine whether or not the recipe can be decrafted/changed in shimmer. The conditions can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="conditions">An array of conditions.</param>
+	public Recipe AddShimmerCondition(params Condition[] conditions) => AddShimmerCondition((IEnumerable<Condition>)conditions);
+
+	public Recipe AddShimmerCondition(Condition condition)
 	{
-		notDecraftable = true;
+		ShimmerConditions.Add(condition);
 		return this;
 	}
 
-	public Recipe CrimsonOnly()
+	/// <summary>
+	/// Adds a collection of conditions that will determine whether or not the recipe can be decrafted/changed in shimmer. The conditions can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="conditions">A collection of conditions.</param>
+	public Recipe AddShimmerCondition(IEnumerable<Condition> conditions)
 	{
-		crimson = true;
+		ShimmerConditions.AddRange(conditions);
 		return this;
 	}
 
-	public Recipe CorruptionOnly()
+	/// <summary>
+	/// Adds every condition from Recipie.Condtions to Recipie.ShimmerContions, checking for duplicates.
+	/// </summary>
+	public Recipe CopyConditionsToShimmer()
```

Comment:
Needs a better name. Maybe `ApplyConditionsAsShimmerConditions`? 

Maybe instead there should be a single function call to add a condition for both crafting and decrafting? `AddCraftingAndDecraftingCondition`?

Reply:
I'll just do both then?

Reply:
I've decided to just change the name as they do essentially the same thing except the new one is 4 extra functions to maintain parity with vanilla